### PR TITLE
minimal: Fix crash due to assertion fail

### LINKED ISSUE #8510: Minimal crash with enabled damage
##### Issue type
 - Bug report
 
##### Minetest version
```
Minetest 5.1.0-dev-e32a630 (Linux)
Using Irrlicht 1.8.4
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_GETTEXT=1
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=0
STATIC_SHAREDIR="."
```

##### OS / Hardware
Operating system: Lubuntu 18.04

Hardware probably does not matter.

##### Summary
The minimal game crashes on world enter if damage is enabled:
```
2019-05-02 12:17:09: ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod '*builtin*' in callback on_joinplayer(): <minetest_dir>/games/minimal/mods/test/init.lua:27: assertion failed!
2019-05-02 12:17:09: ERROR[Main]: stack traceback:
2019-05-02 12:17:09: ERROR[Main]: 	[C]: in function 'assert'
2019-05-02 12:17:09: ERROR[Main]: 	<minetest_dir>/games/minimal/mods/test/init.lua:27: in function 'run_hpchangereason_tests'
2019-05-02 12:17:09: ERROR[Main]: 	<minetest_dir>/games/minimal/mods/test/init.lua:79: in function '?'
2019-05-02 12:17:09: ERROR[Main]: 	<minetest_dir>/builtin/game/register.lua:419: in function <<minetest_dir>/builtin/game/register.lua:399>
```
This has probably something to do with changes in hp reason stuff or whatever.

Probably related: #8494 @ClobberXD 

##### Steps to reproduce
- Start minetest.
- Select minimal game in singleplayer tab.
- Create a new world.
- Enable Damage.
- Select the world and press play.
- Watch how the game crashes with a lua error.

##### Possible Fix
Change this line:
https://github.com/minetest/minetest/blob/e32a630937de69454c741e93979a7889fcddc7a9/games/minimal/mods/test/init.lua#L22
to:
```lua
player:set_hp(12, { a= 234 })
```

### PR DESCRIPTION
**Pull request by ClobberXD**
Closes #8510 (thanks to @DS-Minetest for reporting the crash). Tested; works as it should.

### CODE REVIEW

File: games/minimal/mods/test/init.lua [hunk 1]
```diff
@@ -19,14 +19,14 @@ local function run_hpchangereason_tests(player)
...
-	player:set_hp(10, { a= 234 })
+	player:set_hp(7, { a= 234 })
 	assert(expect == nil)
 
 	expect = { df = 3458973454, type = "fall", from = "mod" }
 	player:set_hp(10, { type = "fall", df = 3458973454 })
 	assert(expect == nil)
 end
-minetest.register_on_player_hpchange(function(player, hp, reason)
+minetest.register_on_player_hpchange(function(_, _, reason)
```
Comment (paramat) [inline games/minimal/mods/test/init.lua]: Is there a benefit to this other than visual cleanliness?

**Comment by paramat** [comment]
Fix is fine, unsure about the codestyle change. I'm not sure if it is our policy to change arguments to '_'.

**Comment by paramat** [comment]
I seem to remember in MTG we are actually adding arguments back in and removing '_', because it makes clearer what arguments are available.
I'm +1 for the PR and will merge it if that codestyle change is removed, otherwise i need to be convinced it's useful.

File: games/minimal/mods/test/init.lua (hunk 1 above)
Comment (ClobberXD) [reply games/minimal/mods/test/init.lua]: AFAIK, no. They're unused variables, and should be `_` as per convention.

Comment (ClobberXD) [reply games/minimal/mods/test/init.lua]: Oh I see this has already been reverted, never mind... :)

**Pull request merged**